src/cdomains/valueDomain.ml

```diff
@@ -1246,7 +1246,18 @@ struct
                 else
                   Invariant.none
               in
-              let i_deref = deref_invariant ~vs ~lval vi offset (Mem c_exp, NoOffset) in
+              let i_deref =
+                match Cilfacade.typeOfLval (Var vi, offset) with
+                | TVoid _ -> Invariant.none (* TODO: Try to determine type of abstract value and introduce appropriate casts *)
```

Comment:
Doesn't this defeat the whole point of `i_deref` because all of our blobs have `void` type? The reason for these dereferenced invariants was precisely to be able to say something about allocated memory that otherwise cannot be referred to.

Reply:
Actually, after checking this again, it does not seem to be an issue anymore, at least with the asserts on blobs being deactivated (with a separate option).